src/lib.rs

```diff
@@ -580,16 +581,16 @@ impl EGraph {
     }
 
     /// find the leader value for a particular eclass
-    pub fn find(&self, value: Value) -> Value {
-        if let Some(sort) = self.get_sort_from_value(&value) {
-            if sort.is_eq_sort() {
-                return Value {
-                    tag: value.tag,
-                    bits: usize::from(self.unionfind.find(Id::from(value.bits as usize))) as u64,
-                };
+    pub fn find(&self, is_eq_sort: bool, value: Value) -> Value {
```

Comment:
This interface change does not seem to make sense, like if the user already knows it's not an eqsort then they should just use not call this function

Reply:
This was a useful forcing for the refactor but I will replace it with a debug assertion

Reply:
Actually, I take this back. If you look at my uses of this function in this PR this is a useful parameter.

Reply:
Sure, I can do that